backend/src/users/users.controller.ts

```diff
@@ -39,10 +47,56 @@ export class UsersController {
   })
   @Delete(':nickname')
   deleteUser(@Param('nickname') nickname: string, @Req() req: Request) {
-    return this.usersService.deleteUser(
+    return this.usersService.deleteUser(nickname, req.user['nickname']);
+  }
+
+  @UseGuards(JwtGuard)
+  @ApiBearerAuth()
+  @ApiParam({
+    name: 'friendNickname',
+    type: String,
+    required: true,
+    description: 'Nickname of the friend you are adding'
+  })
+  @ApiParam({
+    name: 'nickname',
+    type: String,
+    required: true,
+    description: 'Nickname of the user you are adding a friend to'
+  })
+  @Post(':nickname/friends/:friendNickname')
+  addFriend(
+    @Param('nickname') nickname: string,
+    @Param('friendNickname') friendNickname: string,
+    @Req() req: Request
+  ) {
+    return this.usersService.addFriend(
       nickname,
-      req.user['nickname'],
-      req.user['avatarPath']
+      friendNickname,
+      req.user['id']
     );
   }
+
+  @UseGuards(JwtGuard)
+  @ApiBearerAuth()
+  @Delete(':nickname/friends/:friendNickname')
+  deleteFriend(
+    @Param('nickname') userNickname: string,
+    @Param('friendNickname') friendNickname: string,
+    @Req() req: Request
+  ) {
+    return this.usersService.deleteFriend(
+      userNickname,
+      friendNickname,
+      req.user['id']
+    );
+  }
+
+  @UseGuards(JwtGuard)
+  @ApiBearerAuth()
+  @Get(':nickname/friends')
+  // TODO: return friend object instead of only nickname
```

Comment:
What's a friend object?
Isn't it a user?

Reply:
Changed after last commit, this comment is depreciated